src/WorkBC.ElasticSearch.Indexing/Services/XmlParsingServiceFederal.cs

```diff
@@ -86,6 +86,17 @@ public ElasticSearchJob ConvertToElasticJob(string federalXml, bool isFrench = f
                         noc2021 = 0;
                     }
                 }
+                //Special case for NOC-387: The parser should convert any incoming NOC 00011-00015 into NOC 00018
+                //before attempting to locate it in the database. 
+
+                List<short> specialNocs = new List<short>();
+                short[] input = { 0012, 0013, 0014, 0015, 0016 };
```

Comment:
Although I understand it _should_ be equivalent functionally, I was expecting you to examine the incoming `xmlJobNode["noc2021"]` to match against the range 00011-00015.

If you can think of any reason why they wouldn't be equivalent, let's discuss more. Otherwise, looks good to me.

Reply:
Also, I would prefer to see a unit test for this case, please.

Reply:
> Although I understand it _should_ be equivalent functionally, I was expecting you to examine the incoming `xmlJobNode["noc2021"]` to match against the range 00011-00015.
> 
> If you can think of any reason why they wouldn't be equivalent, let's discuss more. Otherwise, looks good to me.

Hi Karim,

The check has to be on NOC 2016 codes because there is a validation condition while parsing the Job XML for Noc2021 codes that they should be present in the NOCCodes2021 table. In our SSOT_NOCS.json file, there is no corresponding data for NOC 2021 values of 00011-00015 range and so they will be assigned as 0 upon parsing. Below is the existing validation code.
<img width="461" alt="NOC2021Id parser" src="https://github.com/bcgov/workbc-jb/assets/63005591/22c4a2b3-a40b-42c4-8e8e-6f561743eb3e">

Reply:
You can check for `xmlJobNode["noc2021"]` before you make that check above. 

Again, the important part if being confident that 0012-0016 is always equivalent to 00011-00015 in the Federal feed.

Reply:
> Also, I would prefer to see a unit test for this case, please.

Added a unit test.